### PR TITLE
Fix constrained match in record

### LINKED ISSUE #674: Regression: ast changed
```
let _ =
  { a
  ; b =
      (match b with
         | `A -> A
         | `B -> B
         | `C -> C : c)
  ; c = c
  }
```

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #674 (no regression with test_branch)

### CODE REVIEW

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -1998,13 +1998,18 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
            | Pexp_constraint ({pexp_desc= Pexp_pack _}, _) ->
                general_case ()
            | Pexp_constraint (e, t) when List.is_empty f.pexp_attributes ->
+               let parens =
```
Comment (hhugo) [inline src/Fmt_ast.ml]: This logic should live in `Ast.parenze_exp`

**Comment by hhugo** [comment]
introduced in https://github.com/ocaml-ppx/ocamlformat/pull/565

**Comment by hhugo** [comment]
I gave it a try

**Comment by gpetiot** [comment]
No regression with test_branch, looks good to me.

**Pull request merged**